### ISSUE TITLE
Transfer change allocation not being registered

### ISSUE THREAD

**Issue by claudiosdc** [rgb-org/rgb-node#129]
When doing an asset transfer operation, the asset change amount is not being allocated to the designated transaction output passed via the `-a` option to the `fungible transfer` command.

This can be verified by issuing the `fungible list -l` command, at the sending RGB node, after the transfer is complete, and noticing that the asset change allocation is not listed under the `known_allocations` entry.

A complete example of such operation can be found at issue #128.

I have also done some research and found the potential cause for this issue. Please see my comments at https://github.com/rgb-org/rgb-node/issues/128#issuecomment-766827213.

Finally, I would like to point out that, even though this misbehavior has been verified in version 0.2.1 of RGB node, base on my code review, it should still be present in the latest version of RGB node.

**Comment by rajarshimaitra** [comment]
Hi @claudiosdc , thanks for reporting the issue. I have been able to reproduce it and your analysis seems to be correct. 

I am working on the fix. If you have already fixed it, feel free to open a PR, I will review and test it.

**Comment by claudiosdc** [comment]
> Hi @claudiosdc , thanks for reporting the issue. I have been able to reproduce it and your analysis seems to be correct.
> 
> I am working on the fix. If you have already fixed it, feel free to open a PR, I will review and test it.

Hi @rajarshimaitra, thank you for your reply.

I have not opened a pull request because I am not sure how to best resolve this. I have been thinking about it, and it seems to me that the produced consignment should be validated before adding the asset change allocation to the asset status of the sending RGB node to, at a minimum, make sure that the witness transaction spends the UTXOs containing the single-use seals of the allocations being spent by the asset transfer, much in the same way that it is expected that the receiving end first validates it (via the `fungible validate` command) before accepting the asset transfer (via de `fungible accept` command).

Does that make sense?

**Comment by rajarshimaitra** [comment]
Yes @claudiosdc , that does make sense. 

Though I am also not sure what extent of validation we would need for it to properly make sense. The whole witness transaction can't be validated because rgb-node does not contain private-key information.

And probably it needs more thinking. I am myself trying to understand the validation logic next.

Although for the purpose of this issue and the demo it does seem like a simple bug. Testers would expect to see the asset allocation updated in the sample example. And that seems like a small change. The cache update is missed in `transfer` call.

So I propose we can close this one (after the fix) for now, and continue with the validation discussion here https://github.com/rgb-org/rgb-node/issues/130 

cc: @dr-orlovsky

**Comment by claudiosdc** [comment]
> So I propose we can close this one (after the fix) for now, and continue with the validation discussion here #130

@rajarshimaitra, I am afraid that issue #130 is unrelated to this one, and, as a matter of fact, I have already opened a pull request (#131) to fix that issue.

**Comment by dr-orlovsky** [comment]
The sending party needs to do accept procedure separately of transfer operation, otherwise, if the corresponding witness transaction do not get mined the append-only stash will never see the asset as not being spent.

I think the proper fix is to update the demo and clearly state about this procedure in the README

**Comment by claudiosdc** [comment]
> The sending party needs to do accept procedure separately of transfer operation, otherwise, if the corresponding witness transaction do not get mined the append-only stash will never see the asset as not being spent.
> 
> I think the proper fix is to update the demo and clearly state about this procedure in the README

Yes, I agree. However, as we have discussed it during the dev call yesterday, the code needs to be adjusted to allow for the acceptance of the consignment by the sending party (so that the asset change allocation is properly recorded).

**Comment by dr-orlovsky** [comment]
Solved with https://github.com/rgb-org/rgb-node/pull/138
